### DISCUSSION TITLE
I don't know how to compile tailwindcss in filament, can anyone help me?

### DISCUSSION THREAD

**Discussion by shebaoting** [filamentphp/filament#3250] [General]
I don't know how to compile tailwindcss in filament, can anyone help me?

I followed the documentation exactly. https://filamentphp.com/docs/2.x/admin/appearance#building-themes

```
npm install tailwindcss @tailwindcss/forms @tailwindcss/typography autoprefixer tippy.js --save-dev
```
```
npx tailwindcss init
```

`tailwind.config.js`
```
const colors = require('tailwindcss/colors')

module.exports = {
    content: [
        './resources/**/*.blade.php',
        './vendor/filament/**/*.blade.php',
    ],
    darkMode: 'class',
    theme: {
        extend: {
            colors: {
                danger: colors.rose,
                primary: colors.blue,
                success: colors.green,
                warning: colors.yellow,
            },
        },
    },
    plugins: [
        require('@tailwindcss/forms'),
        require('@tailwindcss/typography'),
    ],
}

```

`vite.config.js`

```
import { defineConfig } from 'vite';
import laravel from 'laravel-vite-plugin';

export default defineConfig({
    plugins: [
        laravel({
            input: [
                'resources/css/app.css',
                'resources/js/app.js',
                'resources/css/filament.css',
            ],
            refresh: true,
        }),
    ],
});
```

In /resources/css/filament.css, import Filament's vendor CSS:

```
@import '../../vendor/filament/filament/resources/css/app.css';
```

Now, I want to change the background colour of the Admin panel

I took this file
`vendor/filament/filament/resources/views/components/layouts/base.blade.php`
and copied it to
`resources/views/vendor/filament/components/layouts/base.blade.php`

I then changed the class of the body to bg-[#fbf8f]

```
    <body @class([
        'bg-[#fbf8f] text-gray-900 filament-body',
        'dark:text-gray-100 dark:bg-gray-900' => config('filament.dark_mode'),
    ])>
```

```
npm run dev
```
After modifying the page content, `bg-[#fbf8f]` is not compiled successfully. What can I do please?

**Comment by uadoronin** [comment]
You should rewrite an already prepared class, in your filament.css.
After
`@import '../../vendor/filament/filament/resources/css/app.css';`
Add
`.filament-body {
    @apply bg-danger-600;
}`

**Reply by danharrin** [reply]
Yes, please do not override views, ever. Unless they're specifically in the documentation, like the brand logo

**Reply by danharrin** [reply]
This CSS solution is much better

**Reply by Hegabovic** [reply]
how did you know that that this class exists? for example if i want to add my own custom color for the primary color , how should i approach that? 

in `tailwind.config.js`
i want to to change the 
```
colors: {
                danger: colors.rose,
                primary: colors.blue,
                success: colors.green,
                warning: colors.yellow,
            },
```
i want to make primary: with specific color that doesn't exist in `tailwindcss/colors`

**Reply by danharrin** [reply]
Please check the Tailwind docs, you can define each shade of the colour in a JavaScript object

**Reply by Hegabovic** [reply]
thanks a lot

**Comment by hamzadevtech1** [comment]
I have a question... I have used the custom theme and did what is mentioned in filament v3 docs... The theme works and my primary and other colors are working too... the problem is Filament is still getting the button colors from a variable of Color pallete class. However, the left menu is picking up my primary color like text-primary-600 working fine... but buttons on right side, like create, edit buttons... still showing default filament v3 color

**Reply by uadoronin** [reply]
https://filamentphp.com/docs/3.x/panels/themes
